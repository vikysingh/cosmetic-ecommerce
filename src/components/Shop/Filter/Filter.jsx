import React from "react"

import { Typography } from "antd"

import globalStyles from "../../../styles/Globals.module.css"
import styles from "./Filter.module.css"

const { Title } = Typography

export default function Filter({ title, filterList }) {
    return <div className={styles.FILTER} >
        <Title level={3}> {title} </Title>
        <div className={globalStyles.flexRowAroundCenter} >
            {
                filterList.map(item => <label key={item.value} > { item.name } </label>)
            }
        </div>
    </div>
}