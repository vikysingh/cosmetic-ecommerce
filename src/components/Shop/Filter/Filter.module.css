.FILTER {
    width: 95%;
    margin-left: 5%;
    overflow: hidden;
    padding: 20px 0;
}

.FILTER h3 {
    margin-left: 20px;
    margin-bottom: 5px;
}

.FILTER div {
    flex-wrap: wrap;
    justify-content: start;
}

.FILTER label {
    width: auto;
    height: auto;
    padding: 5px 10px;
    margin: 5px;
    background-color: #fff;
    cursor: pointer;
}