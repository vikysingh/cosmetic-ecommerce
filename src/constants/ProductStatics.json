{
    "productType": [
        "Blush",
        "Bronzer",
        "Eyebrow",
        "Eyeliner",
        "Eyeshadow",
        "Foundation",
        "Lip Liner",
        "Lipstick",
        "Mascara",
        "Nail polish"
    ],
    "brands": [
        "Benefit",
        "Covergirl",
        "Dalish",
        "Essie",
        "Fenty",
        "Glossier",
        "Iman",
        "L'oreal",
        "Maybelline",
        "Orly"
    ]
}