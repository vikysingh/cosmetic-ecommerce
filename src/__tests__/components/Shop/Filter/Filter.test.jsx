import React from "react"
import Filter from "../../../../components/Shop/Filter/Filter"

import { shallow } from "enzyme"
import ProductStatics from "../../../../constants/ProductStatics.json"

test('Should render Filter from Shop', () => {
    const wrapper = shallow(<Filter title="Products" filterList={ProductStatics.productType} type="products" />)

    expect(wrapper).toMatchSnapshot()
})