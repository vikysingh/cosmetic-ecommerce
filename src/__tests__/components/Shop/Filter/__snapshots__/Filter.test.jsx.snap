// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Filter from Shop 1`] = `
<div
  className="FILTER"
>
  <Title
    level={3}
  >
     
    Products
     
  </Title>
  <div
    className="flexRowAroundCenter"
  >
    <label
      key="Blush"
      onClick={[Function]}
    >
      Blush
       
    </label>
    <label
      key="Bronzer"
      onClick={[Function]}
    >
      Bronzer
       
    </label>
    <label
      key="Eyebrow"
      onClick={[Function]}
    >
      Eyebrow
       
    </label>
    <label
      key="Eyeliner"
      onClick={[Function]}
    >
      Eyeliner
       
    </label>
    <label
      key="Eyeshadow"
      onClick={[Function]}
    >
      Eyeshadow
       
    </label>
    <label
      key="Foundation"
      onClick={[Function]}
    >
      Foundation
       
    </label>
    <label
      key="Lip Liner"
      onClick={[Function]}
    >
      Lip Liner
       
    </label>
    <label
      key="Lipstick"
      onClick={[Function]}
    >
      Lipstick
       
    </label>
    <label
      key="Mascara"
      onClick={[Function]}
    >
      Mascara
       
    </label>
    <label
      key="Nail polish"
      onClick={[Function]}
    >
      Nail polish
       
    </label>
  </div>
</div>
`;
